presto-spi/src/main/java/com/facebook/presto/spi/ConnectorPageSource.java

```diff
@@ -62,4 +65,9 @@
     @Override
     void close()
             throws IOException;
+
+    default CompletableFuture<?> isBlocked()
```

Comment:
Write a few words of comment about how's that suppose to work, it contradicts with intuition about `is-functions` returning `boolean`. Especially `isFinished` above returns `boolean`.